### DISCUSSION TITLE
TFT_eSPI@2.5.22 doesn't compile

### DISCUSSION THREAD

**Discussion by JayNaire** [Bodmer/TFT_eSPI#2437] [Compatible support libraries]
linux->vscode->platformio:
platform = espressif32
board = esp32doit-devkit-v1
framework = arduino

22Feb23
bodmer/TFT_eSPI@2.5.21 - **compiles fine**
bodmer/TFT_eSPI@2.5.22 - **barfs**:
```
In file included from .pio/libdeps/esp32doit-devkit-v1/TFT_eSPI/TFT_eSPI.cpp:24:0:
.pio/libdeps/esp32doit-devkit-v1/TFT_eSPI/Processors/TFT_eSPI_ESP32.c: In member function 'bool TFT_eSPI::initDMA(bool)':
.pio/libdeps/esp32doit-devkit-v1/TFT_eSPI/Processors/TFT_eSPI_ESP32.c:803:3: error: 'spi_bus_config_t' has no non-static data member named 'data4_io_num'
   };
```
That's all I know! :)

**Comment by Bodmer** [comment]
Have fixed, thanks for reporting this.

**Reply by JayNaire** [reply]
Thank you, you can't complain at the service here!
Sadly though, bodmer/TFT_eSPI@2.5.22 still gives the same error for me. (it's not a pressing concern, 5.2.21 does all I need at the moment.)

**Reply by Bodmer** [reply]
I am guessing:
1. You are using a very old ESP32 board package (1.0.x) to get that error. Version 2.0.x onwards won't generate that error, 2.0.7 is the latest.
2. You have not downloaded the new master direct from Github!

**Reply by JayNaire** [reply]
Many thanks, you guessed correctly!

1. I've no idea what version the board package is. I just set platformio.ini -> board = esp32doit-devkit-v1 and off we went.
2. No I haven't! I assumed PIO does that for me. It seems to do everything else

 ---  time passes -- then: **Gasp! It works!**

For posterity: vscode->PIO Sidebar->Dependencies->Outdated <listed shockingly old espressif32/xtensa packages>
Fix: **vscode->PIO Sidebar->Dependencies->Update**

Thank you again Mr. B.

Bonus: I'm willing to bet some of the examples I tried that failed due to LittleFS dependencies will probably work now.
Added bonus: From watching stuff get updated I'm getting the impression that I just might able to use my Olimex OpenOCD JTAG dongle with ESP32. Who knew?
